Add a troubleshooting tips section to ServiceNow Service Catalog documentation. (#748)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/service-now/service-catalog-terraform/troubleshooting-tips.mdx

---
page_title: Troubleshooting Tips - ServiceNow Service Catalog Integration - HCP Terraform
description: Troubleshooting tips for ServiceNow Service Catalog Integration.
---

# HCP Terraform ServiceNow Service Catalog Integration Troubleshooting Tips. 

This page offers troubleshooting tips for common issues with the ServiceNow Service Catalog Integration for HCP Terraform. 
It also provides instructions on how to find and read logs to diagnose and resolve issues.

## Find logs

Logs are crucial for diagnosing issues. You can find logs in ServiceNow in the following places:

### Workflow logs
You can find workflow logs  by clicking on the RITM number on a failed ticket to open the request item. 
Scroll down to **Related Links > Workflow Context** and open the **Workflow Log** tab. 

### Flow logs
You can find workflow logs by clicking on the RITM number on a failed ticket to open the request item . 
Scroll down to **Related Links > Flow Context > Open Context Record** and find the **Flow engine log entries** tab. 

> ### Application logs
> If workflow or flow logs aren't sufficient, navigate to **All > System Log > Application Logs.**
> Filter the records by setting the **Application** to "Terraform" for better visibility. 
> Search for logs around the time your issue occurred and you can often find HTTP status codes and detailed error messages.

Reviewer: ```suggestion
### Application logs

To find application logs, navigate to **All > System Log > Application Logs.**
Set the **Application** filter to "Terraform". 
Search for logs around the time your issue occurred. Some records include HTTP status codes and detailed error messages.
```

> ### Outbound requests
> For stats on outgoing API calls from your ServiceNow instance to any external endpoint, including HCP Terraform, navigate  to **All > System Logs > Outbound HTTP Requests.** 
> Customize the table view by adding columns like "URL," "URL Path," and "Application scope." 
> Logs from the Catalog app are marked with the "x_325709_terraform" scope. 
> Use filters and grouping to analyze trends effectively.

Reviewer: ```suggestion
### Outbound requests

ServiceNow logs all outgoing API calls, including calls to HCP Terraform. To view the log of outbound requests, navigate  to **All > System Logs > Outbound HTTP Requests.** 
To customize the table view, add columns like "URL," "URL Path," and "Application scope." 
Logs from the Catalog app are marked with the `x_325709_terraform` scope.
```

## Enable email notifications

> To receive updates on your requested item tickets, you can enable email notifications by:

Reviewer: ```suggestion
To enable email notifications and receive updates on your requested item tickets:
```

> 1. Logging into your ServiceNow instance as an administrator. 
>    **Click System Properties > Email Properties** to configure email notifications. 
>    In the **Outbound Email Configuration** panel, select **Yes** next to the check-box with the email to which you would like these requests to be sent.
> 
> 1. To ensure you have relevant notifications configured in your instance, Navigate to **System Notification > Email > Notifications.**
>    Search for "Request Opened" and "Request Item Commented" and ensure they are activated.
> 
> 1. Reaching out to ServiceNow customer support if you run into any issues with the global configurations.

Reviewer: ```suggestion
1. Log in to your ServiceNow instance as an administrator. 
1. **Click System Properties > Email Properties**. 
1. In the **Outbound Email Configuration** panel, select **Yes** next to the check-box with the email that ServiceNow should send notifications to.

To ensure you have relevant notifications configured in your instance: 

1. Navigate to **System Notification > Email > Notifications.**
1. Search for "Request Opened" and "Request Item Commented" and ensure they are activated.

Reaching out to ServiceNow customer support if you run into any issues with the global configurations.
```

## Common problems

This section details frequently encountered issues and how they can be resolved. 

### Failure to create a workspace

> 
> If you order the "create a workspace" catalog item and  nothing happens in ServiceNow and HCP Terraform does not create a workspace then there are several possible reasons why: 

Reviewer: ```suggestion

If you order the "create a workspace" catalog item and nothing happens in ServiceNow and HCP Terraform does not create a workspace, we recommend you check the following: 
```

> 1. Bad Credentials such as an incorrect token, hostname or organization name.   
>    1. Make sure to use a **Team API Token**. This can be found in HCP Terraform under "API Tokens". 
>    1. Make sure the team API token has the correct permissions. 
>    1. Double-check your organization name by copying and pasting it from HCP Terraform or Terraform Enterprise.
>    1. Double-check your host name. 
>    1. Make sure you created your team API token  in the same organization you are using 
>    1. Test your configuration. First click **Update** to process any changes then **Test Config to make sure the connection is working. 
> 1. Wrong VCS Configuration
>    1. **Identifier** field should not have any spaces. The format is: username/repo_name.
>    1. **Name** can be anything, but it is better to avoid special characters as per naming convention.
>    1. Double-check the OAuth token ID in your HCP Terraform/Terraform Enterprise settings. In your HCP Terraform, navigate to **Organization > Settings > Provider.**

Reviewer: ```suggestion
Ensure your HCP Terraform token, hostname, and organization name is correct.

1. Make sure to use a **Team API Token**. This can be found in HCP Terraform under "API Tokens". 
1. Ensure the team API token has the correct permissions. 
1. Double-check your organization name by copying and pasting it from HCP Terraform or Terraform Enterprise.
1. Double-check your host name. 
1. Make sure you created your team API token  in the same organization you are using 
1. Test your configuration. First click **Update** to process any changes then **Test Config to make sure the connection is working.
  
Verify your VCS configuration.

1. The **Identifier** field should not have any spaces.  The ServiceNow Service Catelog Integration requires that you format repository names in the `username/repo_name` format.
1. The **Name** can be anything, but it is better to avoid special characters as per naming convention.
1. Double-check the OAuth token ID in your HCP Terraform/Terraform Enterprise settings. To retrieve your OAuth token ID, navigate to your HCP Terraform organization's settings page, then click **Provider** in the left navigation bar under **Version Control**.
```

### Failure to successfully order any catalog item

After placing an order for any catalog item, navigate to the comments section in the newly created RITM ticket. 

> The latest comment will contain a response from HCP Terraform.

Reviewer: ```suggestion
The latest comment contains a response from HCP Terraform.
```

### Frequency of comments and outputs

> When you place an order in the  Terraform Catalog, Service Now submits and processes the order then attaches  additional comments  to the order to indicate whether HCP Terraform successfully created the workspace.

Reviewer: ```suggestion
When you place an order in the Terraform Catalog, ServiceNow submits and processes the order, then attaches additional comments to the order to indicate whether HCP Terraform successfully created the workspace.
```

> By default, Service Now pings HCP Terraform every 5 minutes for the latest status of the Terraform run. 
> Meaning, if Service Now has processed a  request  before a ping,  it will not show any comments until the next ping. 

Reviewer: ```suggestion
By default, ServiceNow polls HCP Terraform every 5 minutes for the latest status of the Terraform run. ServiceNow does not show any comments until the next ping. 
```

> To increase the responsiveness and have Service Now post comments  faster, you can decrease the ping interval. 
> To do this, navigate to **All > Flow designer**  and filter the records by setting the **Application** to "Terraform". 
> Under the **name** column click  **Worker Poll Run State**. Click on the trigger and adjust the interval from 5  minutes  to your desired schedule. Click **Done > Save > Activate** to save your changes.

Reviewer: ```suggestion
To configure ServiceNow to poll HCP Terraform more frequently:

1. Navigate to **All > Flow designer**.
1. Set the **Application** filter to **Terraform**.
1. Under the **Name** column click  **Worker Poll Run State**.
1. Click on the trigger and adjust the interva to your desired schedule. 
1. Click **Done > Save > Activate** to save your changes.
```

### Using no-code modules feature

> There are a few possible reasons  why the "provisioning of no code modules and deploying resources" catalog item fails to produce anything  in Service Now and does not create a no-code workspace in HCP Terraform. 
> 1. A [HCP Plus tier](/products/terraform/pricing) is required to use this feature. 
> 1. Ensure the name you enter for your no-code module in the catalog user form matches the no-code module HCP Terraform creates. 

Reviewer: ```suggestion
If ServiceNow fails to deploy a no-code module catalog item, verify the following:

1. Ensure that your HCP Terraform organization has an [HCP Plus tier](https://www.hashicorp.com/products/terraform/pricing) subscription. 
1. Ensure the name you enter for your no-code module in the catalog user form matches the no-code module in HCP Terraform. 
```

### Updating no-code workspaces

If the “update no-code workspace” catalog item returns the output message “No update has been made to the workspace”, then you have not upgraded your no-code module in HCP Terraform. 

### Application Scope

> If you are making customizations and you  encounter unexpected issues, make sure to change the scope from **Global** to **Terraform** and recreate your customized items in the **Terraform scope**. 
> For additional instructions on customizations, read more on the [Example Customizations page](/terraform/cloud-docs/integrations/service-now/service-catalog-terraform/example-customizations).

Reviewer: ```suggestion
If you are making customizations and you encounter unexpected issues, make sure to change the scope from **Global** to **Terraform** and recreate your customized items in the **Terraform scope**. 
For additional instructions on customizations, refer to the [example customizations](/terraform/cloud-docs/integrations/service-now/service-catalog-terraform/example-customizations) documentation.
```

### MID server  

> If you are using a MID server in your configuration, always check the connectivity by using the **Test Config** button on the configurations page. 
> Additionally, when ServiceNow provisions a MID server, navigate to **MID Servers > Servers** to check if the status is “up” and “validated”.

Reviewer: ```suggestion
If you are using a MID server in your configuration, check the connectivity by using the **Test Config** button on the configurations page. 
Additionally, when ServiceNow provisions a MID server, navigate to **MID Servers > Servers** to check if the status is “up” and “validated”.
```
